> There are some build errors:
> 
> ```
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(395,9): error CS0103: The name 'Viewco' does not exist in the current context [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-ios18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(397,22): error CS0103: The name 'navRenderer' does not exist in the current context [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-ios18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(394,29): error CS8600: Converting null literal or possible null value to non-nullable type. [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-ios18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(402,22): error CS8600: Converting null literal or possible null value to non-nullable type. [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-ios18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(395,9): error CS0103: The name 'Viewco' does not exist in the current context [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-maccatalyst18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(397,22): error CS0103: The name 'navRenderer' does not exist in the current context [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-maccatalyst18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(394,29): error CS8600: Converting null literal or possible null value to non-nullable type. [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-maccatalyst18.5]
> C:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellPageRendererTracker.cs(402,22): error CS8600: Converting null literal or possible null value to non-nullable type. [C:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net10.0-maccatalyst18.5]
>     8 Error(s)
> ```

Fixed

--- a/src/Controls/src/Core/Compatibility/iOS/Extensions/ToolbarItemExtensions.cs
+++ b/src/Controls/src/Core/Compatibility/iOS/Extensions/ToolbarItemExtensions.cs
@@ -29,15 +29,20 @@ namespace Microsoft.Maui.Controls.Compatibility.Platform.iOS
 
 		internal static SecondarySubToolbarItem ToSecondarySubToolbarItem(this ToolbarItem item)
 		{
+			var weakItem = new WeakReference<ToolbarItem>(item);
+
 			var action = UIAction.Create(item.Text, null, null, _ =>
 			{
-				if (item is IMenuItemController menuItemController)
+				if (weakItem.TryGetTarget(out var targetItem))
 				{
-					menuItemController.Activate();
-				}
-				else
-				{
-					item.Command?.Execute(item.CommandParameter);
+					if (targetItem is IMenuItemController menuItemController)
+					{
+						menuItemController.Activate();
+					}
+					else
+					{
+						targetItem.Command?.Execute(targetItem.CommandParameter);
+					}
 				}
 			});
 

